File: Algorithms/BinarySearch/binary_search.go
package main

import (
	"fmt"
)

func binarySearch(value int, array []int) int {
 
    low := 0
    high := len(array) - 1
 
    for low <= high{
        median := (low + high) / 2
 
        if array[median] < value {
            low = median + 1
        }else{
            high = median - 1
        }
    }
 
    if low == len(array) || array[low] != value {
        return -1
    }
 
    return low 
}

func main(){
    items := []int{5, 7, 11, 22, 33, 47, 58, 65}
    fmt.Println(binarySearch(11, items))
}

File: Algorithms/BinarySearch/binary_search.js
const binarySearch = ( list, item ) => {
    let low = 0;
    let high = list.length - 1;

    while ( low <= high ) {
        let mid = Math.floor((low + high) / 2);
        let guess = list[mid];

        if ( guess === item ) {
            return mid;
        } else if ( guess > item ) {
            high = mid - 1;
        } else {
            low = mid + 1;
        }
    }

    return null;
};

const myList = [1, 3, 5, 7, 9];

console.log( binarySearch( myList, 3 ) );
console.log( binarySearch( myList, -1 ) );
